torch/distributed/_sharding_spec/api.py

```diff
@@ -0,0 +1,98 @@
+from abc import ABC
+import torch
+from typing import List, Union
+
+from ._internals import is_valid_device
+
+Device = Union[torch.device, int, str]
+
+class PlacementSpec(ABC):
```

Comment:
Why do we need a common superclass here? Will there be any major common attributes or methods?

I feel `List[DevicePlacementSpec]` should just be a data member of `ChunkShardingSpec`, and the relationship is only composition and little inheritance.

Reply:
`DevicePlacementSpec` would be useful in the case where we want to shard an nn.Module. Let's say we have 3 tensors in the module and we want to place them on three different devices. `DevicePlacementSpec` can be used in conjunction with "Composite ShardingSpec" from https://github.com/pytorch/pytorch/issues/55207 to address this.

Reply:
I see, so "sharding" here means not only intra-layer sharding, but also inter-layer sharding (where `DevicePlacementSpec`, i.e., no intra-layer sharding, can be applied).